desktop-src/direct3d9/queries.md

```diff
@@ -181,16 +181,19 @@ pOcclusionQuery->Issue(D3DISSUE_END);
 
 // Force the driver to execute the commands from the command buffer.
 // Empty the command buffer and wait until the GPU is idle.
-while(S_FALSE == pOcclusionQuery->GetData( &numberOfPixelsDrawn, 
+while(S_FALSE == pOcclusionQuery->GetData( &numberOfSamplesShaded, 
                                   sizeof(DWORD), D3DGETDATA_FLUSH ))
     ;
+
+// Divide numberOfSamplesShaded by sample count of the rendertarget
```

Comment:
The division here would not be convenient because there are no surfaces to GetDesc of here, it's not clear which surface is actually bound at this point (only color, only depth-stencil, or both — so we don't know whether we can call GetDepthStencilSurface, and whether we can use GetRenderTarget), and the multisample setting in D3D9 is a enumeration that also contains D3DMULTISAMPLE_NONMASKABLE.

However, switching the order of the clauses in that sentence — starting it with the problem, "To get the number of pixels drawn…", rather than the solution, "Divide…", feels to me like it changes the tone of the comment from being an explanation of what happens after that comment in the code to a suggestion for the reader what to do if they are in the specified situation.

Reply:
Do we know how nonmaskable samples work? It always felt to me like an unwanted feature that MSDN docs say "should be used for antialiasing" but in practice I have **never** seen **anyone** use it in a commercial game.